packages/taquito-michelson-encoder/src/schema/errors.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { InvalidViewParameterError, TaquitoError } from '@taquito/core';
+
+/**
+ *  @category Error
+ *  @description Error that indicates a failure when encoding (transforming JS parameter into JSON Michelson) the parameter of the view
+ */
+export class ParameterEncodingError extends InvalidViewParameterError {
+  constructor(
+    public readonly viewName: string,
+    public readonly sigs: any,
+    public readonly args: any,
+    public readonly cause?: any
+  ) {
+    super(viewName, sigs, args, cause);
+    this.name = 'ParameterEncodingError';
+    this.message = `Could not encode parameter ${JSON.stringify(
+      args
+    )} received for name "${viewName}" expecting one of the following signatures ${JSON.stringify(
+      sigs
+    )}`;
+  }
+}
+
+/**
+ *  @category Error
+ *  @description Error that indicates an invalid on-chain view found on the script
+ */
+export class InvalidScriptError extends TaquitoError {
+  name = 'InvalidScriptError';
+  constructor(public readonly script: any, public readonly reason?: string) {
+    super();
+    let message = `Invalid on-chain view found in the following script.`;
+    if (reason) {
+      message += ` Reason: ${reason}.`;
+    }
+    message += `Script: ${JSON.stringify(script)}`;
+    this.message = message;
+  }
+}
+
+/**
+ *  @category Error
+ *  @description Error that indicates an invalid RPC response being passed or used
+ */
+export class InvalidRpcResponseError extends TaquitoError {
```

Comment:
is there an already existing error that's somewhat similar? might be able to reuse that